server/base/src/main/java/org/apache/accumulo/server/tables/TableManager.java

```diff
@@ -62,23 +63,36 @@ public class TableManager {
   private final String zkRoot;
   private final InstanceId instanceID;
   private final ZooReaderWriter zoo;
-  private ZooCache zooStateCache;
+  private final ZooCache zooStateCache;
 
-  public static void prepareNewNamespaceState(ZooReaderWriter zoo, InstanceId instanceId,
-      NamespaceId namespaceId, String namespace, NodeExistsPolicy existsPolicy)
-      throws KeeperException, InterruptedException {
+  public static void prepareNewNamespaceState(ZooReaderWriter zoo, final PropStore propStore,
+      InstanceId instanceId, NamespaceId namespaceId, String namespace,
+      NodeExistsPolicy existsPolicy) throws KeeperException, InterruptedException {
     log.debug("Creating ZooKeeper entries for new namespace {} (ID: {})", namespace, namespaceId);
     String zPath = Constants.ZROOT + "/" + instanceId + Constants.ZNAMESPACES + "/" + namespaceId;
 
     zoo.putPersistentData(zPath, new byte[0], existsPolicy);
     zoo.putPersistentData(zPath + Constants.ZNAMESPACE_NAME, namespace.getBytes(UTF_8),
         existsPolicy);
     zoo.putPersistentData(zPath + Constants.ZNAMESPACE_CONF, new byte[0], existsPolicy);
+
+    PropCacheKey propKey = PropCacheKey.forNamespace(instanceId, namespaceId);
+    if (!propStore.exists(propKey)) {
+      propStore.create(propKey, Map.of());
```

Comment:
If multiple threads could call this, this code can have race condition.  However not sure of the context in which this particular code is used or if that is a concern.  If it does matter a model of throwing exceptions when it exists (like ZK does and accumulo create table) can avoid race conditions.

Reply:
Currently if multiple writers try to do a create one will succeed the others will fail - but there will only be one created.